Sources/KlaviyoSwift/AppContextInfo.swift

```diff
@@ -30,17 +46,33 @@ struct AppContextInfo {
         "\(osName) \(osVersion)"
     }
 
-    init(excutable: String = DEFAULT_EXECUTABLE,
+    init(executable: String = DEFAULT_EXECUTABLE,
          bundleId: String = DEFAULT_BUNDLE_ID,
          appVersion: String = DEFAULT_APP_VERSION,
          appBuild: String = DEFAULT_APP_BUILD,
+         appName: String = DEFAULT_APP_NAME,
          version: OperatingSystemVersion = DEFAULT_OS_VERSION,
-         osName: String = DEFAULT_OS_NAME) {
-        self.excutable = excutable
+         osName: String = DEFAULT_OS_NAME,
+         manufacturer: String = DEFAULT_MANUFACTURER,
+         deviceModel: String = DEFAULT_DEVICE_MODEL,
+         deviceId: String? = nil) {
+        self.executable = executable
         self.bundleId = bundleId
         self.appVersion = appVersion
         self.appBuild = appBuild
+        self.appName = appName
         self.version = version
         self.osName = osName
+        self.manufacturer = manufacturer
+        self.deviceModel = deviceModel
+
+        if let _deviceId = deviceId {
+            self.deviceId = _deviceId
+        } else if let _deviceId = UserDefaults.standard.string(forKey: DEVICE_ID_STORE_KEY) {
```

Comment:
@ajaysubra and @ndurell let me know if this looks like an appropriate way to save our own device UUID for the lifetime of the app install. Namely wondering if you're cool with using user defaults here. 

I was hesitant to use identifierForVendor since it can be duplicated across apps, if the vendor has multiple apps installed on the device.

Reply:
1. minor nit - the unwrapped variable's name can be the same as the variable name of the optional variable in swift.
2. About using `UserDefaults` I mostly try and avoid using it since it's very specific to storing any default setting the user makes like using Celsius over Fahrenheit etc. But I do see your point about using `identifierForVendor`. Reading apple's documentation on this, seems like they intentionally made `identifierForVendor`  limited to the lifetime of the app for privacy reasons. Should we just remain compliant with what apple recommends just to be safe and not risk user customers apps getting reviewed and flagged for doing something outside of what they recommend? I know I'm probably over thinking this but in my experience apple's review can be nit picky at times.

Also, I'm curious what is the issue if the device ID is same across different apps from the same vendor? I would expect it to be the same since it's the same device no?

P.S - found [this](https://nshipster.com/device-identifiers/)  article interesting especially the part about apple building the DeviceCheck framework for Uber.

Reply:
I guess i could re-consider my data structure on the backend, but I've defined device ID as unique in the data model, so if there's a clear way to wind up with duplicates (e.g. if a company uses us for two different apps), that presents a bit of a problem on the backend right now.

Reply:
Actually i might just be overthinking it.... A company can only have 1 app configured with us now anyway. Whenever we change that, we'll probably have a variety of required changed to the metadata model anyways.

Reply:
And yea, in the future it might be _nice_ to be able to link devices across apps if we had a bigger client with multiple apps. Ok gonna change my approach then.

Reply:
Yeah, that's a good plan. I think the biggest use cases of multi app users are marketing agencies. When I was at constant contact our iOS app supported multi-user experience where marketing agencies would have multiple accounts setup on our app and switch between them. It was a pain the neck to maintain but was a good lesson in building for scale.

Reply:
Yah slightly concerned about this - but at the same time maybe it's ok. In the case where a customer had multiple apps are we ok with this having the same id? I think it would be ok since we also use company id and we only allow one  app per company. If we started to allow multiple apps per company i'm wondering if it means we would have one device entry for both apps.